gprofiler/profilers/perf.py

```diff
@@ -12,14 +12,34 @@
 
 from gprofiler.log import get_logger_adapter
 from gprofiler.merge import ProcessToStackSampleCounters, merge_global_perfs
-from gprofiler.profiler_base import ProfilerBase
+from gprofiler.profilers.profiler_base import ProfilerBase
+from gprofiler.profilers.registry import ProfilerArgument, register_profiler
 from gprofiler.utils import TEMPORARY_STORAGE_PATH, resource_path, run_process
 
 logger = get_logger_adapter(__name__)
 
 PERF_BUILDID_DIR = os.path.join(TEMPORARY_STORAGE_PATH, "perf-buildids")
 
 
+@register_profiler(
+    "Perf",
+    possible_modes=["fp", "dwarf", "smart", "disabled", "none"],
+    default_mode="fp",
+    profiler_mode_argument_help="Run perf with either FP (Frame Pointers), DWARF, or run both and intelligently merge"
+    " them by choosing the best result per process. If 'none' is chosen, do not invoke"
+    " 'perf' at all. The output, in that case, is the concatenation of the results from all"
+    " of the runtime profilers.",
+    profiler_arguments=[
+        ProfilerArgument(
```

Comment:
I admit this does look nice, but find it weird that you need to have this `ProfilerArgument` class which is merely arguments to an `add_argument` call. Any other way to do it more nicely (either providing a function here to do the `add_argument` calls, or ... need to think)

Reply:
I actually like this. It's basically a dataclass that wraps params for `add_argument`. Sounds like something you'd want when wrapping the use of `add_argument` with a list of objects to call with.

Reply:
`namedtuple` will suffice IMO

Reply:
Agree that this way looks better than passing a function object here (which gets called and adds the new arguments)

Reply:
Wrote in another comment why `namedtuple` is not enough

Reply:
k

Reply:
resolving